src/FailureDetectorConfig.md

```diff
@@ -85,3 +88,65 @@ config.setProperty("hazelcast.heartbeat.phiaccrual.failuredetector.sample.size",
 config.setProperty("hazelcast.heartbeat.phiaccrual.failuredetector.min.std.dev.millis", "100");
 [...]
 ```
+
+### Ping Failure Detector
+
+This failure detector is based on `InetAddress.isReachable()`.
+When the JVM process has enough permissions to create RAW sockets, the implementation will choose to rely on ICMP Echo requests, whereas, if not enough permissions, it will fallback on attempting a TCP Echo on port 7. In the latter, both a successful connection or an explicit rejection, will be treated as `Host is Reachable`.
+
+For the Ping Failure Detector to rely **only** on ICMP Echo requests, there are some criteria that need to be met.
+#### Requirements
+- Supported OS, as of Java 1.8 *nix environment support this
+- The Java executable must have the `cap_net_raw` capability. 
+
+To do so, run `sudo setcap cap_net_raw=+ep <JDK_HOME>/jre/bin/java`
+- When running with custom capabilities, the dynamic linker on Linux will reject loading libs from untrusted paths.
+An example of rejected execution `java: error while loading shared libraries: libjli.so: cannot open shared object file: No such file or directory`
+
+To overcome, the `<JDK_HOME>/jre/lib/amd64/jli/` path needs to be added in the `ld.conf`
+`echo "<JDK_HOME>/jre/lib/amd64/jli/" >> /etc/ld.so.conf.d/java.conf && sudo ldconfig`
+- ICMP Echo Requests must not be blocked by the receiving hosts. `/proc/sys/net/ipv4/icmp_echo_ignore_all` set to `0`
+
+i.e. execute `echo 0 > /proc/sys/net/ipv4/icmp_echo_ignore_all`
+
+If any of the above criteria isn't met, then the `isReachable` will always fallback on TCP Echo attempts on port 7.
+
+To use the Ping Failure Detector.
```

Comment:
To be able to use the Ping Failure Detector, please add the following properties in your Hazelcast declarative configuration file: